### PR TITLE
[Fix] app.proto: fix related to container field

### PR DESCRIPTION
**Pull request by Bodong-Yang**
This PR fixes some bugs related to container fields.

* fix the behavior when field descriptor for container field handling default value. It should set the value with empty container wrapper instance, not plain list/dict.
* `MessageMappingContainer` now deprecates the `export_pb` and `MessageMappingContainer` should only use `export_to_pb_msg_mapping_container` to export to protobuf.
* the `_FieldBase` abc now marks `__set__` method as abstracted method, all detailed field descriptor implementations must implement the `__set__` method.

More general tests are implemented to cover most of the cases which we support.

### CODE REVIEW

**Review by obi-t4 — COMMENTED** [review]
Is it possible to add test code for this fix?

**Comment by Bodong-Yang** [comment]
> Is it possible to add test code for this fix?

Sure! I will add a test to test the behavior related to default behavior.

File: otaclient/app/proto/_common.py [hunk 1]
```diff
@@ -232,11 +233,27 @@ def convert(
         return res
 
     def export_pb(self) -> Dict[_K, Any]:
+        warnings.warn(
```
Comment (obi-t4) [inline otaclient/app/proto/_common.py:236]: @Bodong-Yang 
Is this warn required?
I think abort is enough.

Comment (Bodong-Yang) [reply otaclient/app/proto/_common.py:236]: Abort, do you mean directly return?

Comment (obi-t4) [reply otaclient/app/proto/_common.py:236]: No, this function is internal, and if this MessageMapContainer::export_bp is called, it is bug.
So raising exception or something should be done, if I'm undersdanding it correctly.

**Review by obi-t4 — APPROVED** [review]
LGTM :+1:

**Pull request merged**